model/monitorCache/implementation/DeliveryMonitoringData.php

```diff
@@ -119,6 +120,34 @@ public function setDeliveryExecution(DeliveryExecution $deliveryExecution)
         $this->deliveryExecution = $deliveryExecution;
     }
 
+    /**
+     * @return DeliveryExecutionInterface
+     */
+    public function getDeliveryExecution()
+    {
+        return $this->deliveryExecution;
+    }
+
+    /**
+     * {@inheritdoc
+     */
+    public function setDeliveryExecutionContext(DeliveryExecutionContextInterface $context)
+    {
+        $this->update(self::PARAM_EXECUTION_CONTEXT, json_encode($context));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDeliveryExecutionContext()
+    {
+        if (isset($this->data[self::PARAM_EXECUTION_CONTEXT])) {
+            return DeliveryExecutionContext::createFromArray(json_decode($this->data[self::PARAM_EXECUTION_CONTEXT], true));
+        }
+
+        return false;
```

Comment:
What do you think about throwing an exception?

Reply:
As we discussed it's not a required parameter and it will be better to return `null`